p2panda-net-next/src/actors/streams/eventually_consistent.rs

```diff
@@ -0,0 +1,342 @@
+                        .topic_manager_map
+                        .insert(topic_id, (sync_manager_actor.clone(), live_mode));
+
+                    state
+                        .sync_managers
+                        .actor_topic_map
+                        .insert(sync_manager_actor.get_id(), topic_id);
+
+                    EventuallyConsistentStream::new(
+                        state.actor_namespace.clone(),
+                        topic_id,
+                        sync_manager_actor,
+                    )
+                };
+
+                // Ignore any potential send error; it's not a concern of this actor.
+                let _ = reply.send(stream);
+            }
+            ToEventuallyConsistentStreams::Subscribe(topic_id, reply) => {
+                if let Some(from_sync_rx) = state.sync_receivers.get(&topic_id) {
+                    let subscription =
+                        EventuallyConsistentSubscription::new(topic_id, from_sync_rx.resubscribe());
+
+                    let _ = reply.send(Some(subscription));
+                } else {
+                    let _ = reply.send(None);
+                }
+            }
+            ToEventuallyConsistentStreams::Close(topic_id) => {
+                // Tell the gossip actor to unsubscribe from this topic id.
+                state.unsubscribe_from_gossip(topic_id).await?;
+
+                // Drop all senders and receivers associated with the topic id.
+                state.gossip_senders.remove(&topic_id);
+                state.sync_receivers.remove(&topic_id);
+
+                // Drop the sync manager state for this topic id.
+                if let Some((sync_manager, _)) =
+                    state.sync_managers.topic_manager_map.remove(&topic_id)
+                {
+                    state
+                        .sync_managers
+                        .actor_topic_map
+                        .remove(&sync_manager.get_id());
+
+                    // Finish processing all messages in the manager's queue and then kill it.
+                    sync_manager.drain()?;
+                }
+            }
+            ToEventuallyConsistentStreams::InitiateSync(topic_id, node_id) => {
+                if let Some((sync_manager_actor, live_mode)) =
+                    state.sync_managers.topic_manager_map.get(&topic_id)
+                {
+                    sync_manager_actor
+                        .send_message(ToSyncManager::Initiate(node_id, topic_id, live_mode))?;
+                }
+            }
+            ToEventuallyConsistentStreams::EndSync(topic_id, node_id) => {
+                if let Some((sync_manager_actor, _)) =
+                    state.sync_managers.topic_manager_map.get(&topic_id)
+                {
+                    sync_manager_actor.send_message(ToSyncManager::Close(node_id, topic_id))?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn handle_supervisor_evt(
```

Comment:
We handle supervision events for the sync manager actors here. This replicates the approach we take in the root gossip actor (which supervises the underlying gossip session actors).